content/sensu-go/6.1/operations/maintain-sensu/troubleshoot.md

```diff
@@ -379,6 +379,123 @@ Since the `lsb_release` package is not installed, the agent will not be able to
+
+### Symptoms of poor performance
+
+At the default "warn" log level, you may see messages like these from your Sensu backend:
+
+{{< code json >}}
+{"component":"etcd","level":"warning","msg":"read-only range request \"key:\\\"/sensu.io/handlers/default/keepalive\\\" limit:1 \" with result \"range_response_count:0 size:6\" took too long (169.767546ms) to execute","pkg":"etcdserver","time":"..."}
+{{< /code >}}
+
+The above message indicates that a database query ("read-only range request") exceeded a 100 millisecond threshold hard-coded into etcd. Messages like these are helpful in so far as they can alert us to a trend, but occasional warnings of this nature aren't necessarily indicative of a problem.
+
+However, a trend of increasingly long-running database transactions will eventually lead to decreased reliability. You may experience symptoms of these conditions as inconsistent behavior around check execution or updates to configuration fail to be applied as expected.
+
+As the etcd project's [tuning documentation][14] states, etcd is "very sensitive to disk latencies. Since etcd must persist proposals to its log, disk activity from other processes may cause long fsync latencies. [...] etcd may miss heartbeats, causing request timeouts and temporary leader loss."
+
+When Sensu's etcd component doesn't recieve sufficient CPU cycles, or when it's file system can't sustain a sufficient number of IOPS, transactions will begin to timeout, leading to cascading failures:
+
+A message like this one indicates that syncing the etcd database to disk exceeded another threshold:
+
+{{< code json >}}
+{"component":"etcd","level":"warning","msg":"sync duration of 1.031759056s, expected less than 1s","pkg":"wal","time":"..."}
+{{< /code >}}
+
+These subsequent "retrying of unary invoker failed" messages are indicative of failing requests to etcd:
+
+{{< code json >}}
+{"level":"warn","ts":"...","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"endpoint://client-6f6bfc7e-cf31-4498-a564-78d6b7b3a44e/localhost:2379","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
+message repeated 5 times
+{{< /code >}}
+
+In many cases the backend service detects and attempts to recover from errors like these, so you may see a message like this one:
+
+{{< code json >}}
+{"component":"backend","error":"error from keepalived: internal error: etcdserver: request timed out","level":"error","msg":"backend stopped working and is restarting","time":"..."}
+{{< /code >}}
+
+In some scenarios this may result in a crash loop that is difficult to recover from. You may observe that the Sensu backend process continues running, but is not listening for connections on the agent websocket, API or web UI ports. The backend will stop listening on those ports when the etcd database is unavailable.
```

Comment:
```suggestion
This may result in a crash loop that is difficult to recover from.
You may observe that the Sensu backend process continues running but is not listening for connections on the agent websocket, API, or web UI ports.
The backend will stop listening on those ports when the etcd database is unavailable.
```